moto/ec2/responses/security_groups.py

```diff
@@ -86,6 +86,15 @@ def _process_rules_from_querystring(self) -> Any:
                 d = d[subkey]
             d[key_splitted[-1]] = value
 
+        sg_rule_tags = {}
```

Comment:
I think we should process the tag-retrieval inside the `authorize_..`-method itself. It is a toplevel attribute, and arguably unrelated to the other SecurityGroupRule-properties that we're parsing.

Ideally we can just use 
`tags = self._parse_tag_specification().get("security-group-rule", {})`,
similar to how it's done in `create_security_group`. That way we don't have to have any complicated parsing here